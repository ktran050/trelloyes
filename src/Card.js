import "./Card.css";
import React from "react";

class Card extends React.Component {
  render() {
    return (
      <div className="Card">
        <h3>{this.props.title}</h3>
        <p>{this.props.content}</p>
      </div>
    );
  }
}

export default Card;

// The Card component should render markup matching the design.html:
// a div with a class of 'Card' containing
// an h3 for the card's title
// and a p element for the card's content.
// The Card component accepts 2 props: title and content.
// title is a string of the card's title.
// content is a string of the card's content.
// These props will be passed in for each Card from the List component.
