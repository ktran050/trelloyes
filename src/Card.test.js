import React from "react";
import ReactDOM from "react-dom";
import Card from "./Card";
import renderer from "react-test-renderer";
it("renders cards without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Card title="smoke" content="test" />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders cards in the manner expected", () => {
  const tree = renderer
    .create(<Card title="snapshot" content="test" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
