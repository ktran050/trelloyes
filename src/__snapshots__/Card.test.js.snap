// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cards in the manner expected 1`] = `
<div
  className="Card"
>
  <h3>
    snapshot
  </h3>
  <p>
    test
  </p>
</div>
`;
